src/util/discretizer.py

```diff
@@ -62,36 +67,40 @@ def fit(self, X, y=None):
         return self
 
     def transform(self, X):
-        """Transform X using the forward function.
+        """
+        Transform X using the forward function.
         Parameters
         ----------
         X : pandas dataframe containing columns ['userId', 'movieId', 'rating']
         Returns
         -------
         X_out : pandas dataframe containing columns ['userId','rating']
             where rating equals 1 or -1 if user likes the movie or not
-            or NaN if user was not present while fitting the CustomScaler
+            or NaN if user was not present while fitting the RatingDiscretizer
             if more columns were passed to the transform function,
             only 'rating' column is modified and the rest is returned unchanged
         """
-        Xcopy = X.copy()
+        x_copy = X.copy()
         valid_user_ids = set(X["userId"]) & set(self.means.index)
         ratings = X.set_index("userId").rating
-        ratings = np.sign(ratings - self.means[valid_user_ids]).values
-        Xcopy.rating.update(pd.Series(ratings, X.rating.index))
+        # liked = np.sign(ratings - self.means[valid_user_ids]).values
+        liked = ratings - self.means[valid_user_ids]
+        liked = liked > 0
```

Comment:
Im fine with that, but keep in mond that this solution will not propagate NaN's in case userId was not present in the training set. np.sign propagated NaNs - either way such situation should not occur.